Update babylon to 7

danez opened the pull request:
This update babylon to the latest beta of 7.0. This version is the first to include the estree plugin.
~The estree plugin will be also released with a minor 6.x version.~

It looks really good, nearly no changes necessary, only some tests according to raw value are failing.

TODO:

- [x] depends on #157 
- [x] raw values differ from babylon 5 (will fix upstream)
- [x] ~ranges not supported in babylon >= 6 (will investigate upstream)~

MoOx commented:
Amazing. Looking forward to get this!

danez commented:
We probably want to merge this with #137 and make the parser configurable (Maybe similar to prettier)? I think @fkling had already plans for this.

Can also work on this if we can agree how it should work.

//cc @wbinnssmith

fkling commented:
Awesome! Yes, making the parser configurable was the idea because flow doesn't support everything that babylon supports. However, we could probably also just merge this PR to finally get support for new flow types.

How does prettier let you specify parsers?

danez commented:
Prettier has this file: https://github.com/prettier/prettier/blob/master/src/parser.js
and in the options it can be specified which parser to use. So still limited to flow/babylon, but one can choose between these two. They probably did this, because babylon (still) does not understand all flow features.

I would wait with merging till the next 6.x release is out (today/tomorrow), which has the same estree functionality as this beta version. (https://github.com/babel/babylon/blob/changelog-6.16/CHANGELOG.md)

fkling commented:
Ah I see. Yeah, that's exactly what I had in mind (we do pretty much the same thing in jscodeshift).

Thank you for this!

fkling commented:
So, it's not coming in v6.16.0?

danez commented:
The problem is, that babylon 6 also renamed some flow nodes for some odd reason. We now renamed them back to the same names that the flow-parser uses, but this change will only land in 7.0

pascalduez commented:
This should allow us the use common imports for modules and types, which currently crashes:
```js
import { myModule, type myType } from 'my-module';
```

fkling commented on src/utils/__tests__/resolveToValue-test.js:
```diff
@@ -49,7 +49,7 @@ describe('resolveToValue', () => {
   });
 
-  it('handles SpreadProperties properly', () => {
+  it('handles SpreadPElements properly', () => {
```
Tiny typo here `SpreadElements`

fkling commented:
Should we just merge this? I have no problem using a beta version. We could release it as a beta/next version as well, so people could start testing it.

fkling commented:
I merged this into the `v3-dev` branch and will release `react-docgen@next` shortly. Even though switching to v7 shouldn't break anything, it's a big enough change that a major version bump seems appropriate.

danez commented:
Cool thanks
